pkg/operator/ceph/cluster/controller.go

```diff
@@ -1190,8 +1192,24 @@ func populateExternalClusterInfo(context *clusterd.Context, namespace string) *c
 			time.Sleep(externalConnectionRetry)
 			continue
 		} else {
-			logger.Infof("found the cluster info to connect to the external cluster. mons=%+v", clusterInfo.Monitors)
-			break
+			// If an admin key was provided we don't need to load the other resources
+			// Some people might want to give the admin key
+			// The necessary users/keys/secrets will be created by Rook
+			if clusterInfo.AdminSecret != "admin-secret" {
```

Comment:
Let's define a const for "admin-secret" if we're treating it as a special value.

Reply:
It's already a const, just missed in this iteration.